src/user/user.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { Role } from 'src/shared/enums/user-role.enum';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { User, UserDocument } from './schemas/user.schema';
+
+@Injectable()
+export class UserService {
+  constructor(@InjectModel(User.name) private userModel: Model<UserDocument>) {}
+  
+  async create(createUserDto: CreateUserDto) {
+    const user = new this.userModel(createUserDto);
+    
+    return user.save();
+  }
+
+  async findAll() {
+    return this.userModel.find().exec();
+  }
+
+  async findOne(options: Partial<Omit<User, 'password'>>) {
+    return this.userModel.findOne(options).exec();
+  }
+
+  async findByRole(role: Role) {
+    return this.findOne({ role });
+  }
+
+  async findById(id: string) {
+    return this.findOne({ _id: id });
+  }
+
+  async update(id: string, updateUserDto: UpdateUserDto) {
```

Comment:
You could destructure the `updateUserDto` at this level. You don't need to do it beneath

Reply:
For me it is more readable this way :)